tests/bsim/bluetooth/host/misc/sample_test/src/dut.c

```diff
@@ -0,0 +1,167 @@
+					      (BT_GATT_CHRC_READ | BT_GATT_CHRC_WRITE |
+					       BT_GATT_CHRC_NOTIFY | BT_GATT_CHRC_INDICATE),
+					      BT_GATT_PERM_READ | BT_GATT_PERM_WRITE, NULL, NULL,
+					      NULL),
+		       BT_GATT_CCC(ccc_changed, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),);
+
+static void setup(char *message)
+{
+	/* Initialize the bsim framework, and trigger a callback before the
+	 * simulation times out.
+	 */
+	TEST_START(timeout_get(), message);
+
+	bs_sync_init();
+
+	int err = bt_enable(NULL);
+
+	TEST_ASSERT(err == 0, "Can't enable Bluetooth (err %d)\n", err);
+
+	LOG_DBG("Bluetooth initialized.");
+}
+
+/* This is the entrypoint for the DUT.
+ *
+ * This is executed by the `bst_test` framework that comes with the bsim
+ * distribution. The framework selects which "main" function to run as
+ * entrypoint depending on the `-testid=` command-line parameter passed to the
+ * zephyr executable.
+ *
+ * In our case, the `testid` is set to "dut" and `entrypoint_dut()` is mapped to
+ * the "dut" ID in `entrypoints[]`.
+ *
+ * In our case we only have two entrypoints, as we only have a single test
+ * involving two devices (so 1x 2 entrypoints). One can define more test cases
+ * with different entrypoints and map them to different test ID strings in
+ * `entrypoints[]`
+ */
+void entrypoint_dut(void)
+{
+	/* Please leave a comment indicating what the test is supposed to test,
+	 * and what is the pass verdict. A nice place is at the beginning of
+	 * each test entry point. Something like the following:
+	 */
+
+	/* Test purpose:
+	 *
+	 * Verifies that we are able to send a notification to the peer when
+	 * `CONFIG_BT_GATT_ENFORCE_SUBSCRIPTION` is disabled and the peer has
+	 * unsubscribed from the characteristic in question.
+	 *
+	 * Two devices:
+	 * - `dut`: tries to send the notification
+	 * - `peer`: will receive the notification
+	 *
+	 * Procedure:
+	 * - [dut] establish connection to `peer`
+	 * - [peer] discover GATT and subscribe to the test characteristic
+	 * - [dut] send notification #1
+	 * - [peer] wait for notification
+	 * - [peer] unsubscribe
+	 * - [dut] send notification #2
+	 * - [peer] and [dut] pass test
+	 *
+	 * [verdict]
+	 * - peer receives notifications #1 and #2
+	 */
+	int err;
+	bt_addr_le_t peer = {};
+	struct bt_conn *conn = NULL;
+	const struct bt_gatt_attr *attr;
+	uint8_t data[10];
+
+	/* Initialize framework and enable Bluetooth */
+	setup("dut");
+
+	/* Find the address of the peer */
+	err = bt_testlib_scan_find_name(&peer, CONFIG_BT_DEVICE_NAME);
+	TEST_ASSERT(!err, "Failed to start scan (err %d)\n", err);
+
+	/* Create a connection using that address */
+	err = bt_testlib_connect(&peer, &conn);
+	TEST_ASSERT(!err, "Failed to initiate connection (err %d)\n", err);
+
+	LOG_DBG("connected");
+
+	LOG_INF("wait until peer subscribes");
+	UNSET_FLAG(is_subscribed);
+	WAIT_FOR_FLAG(is_subscribed);
+
+	/* prepare data for notifications */
+	attr = &test_gatt_service.attrs[2];
```

Comment:
Should consider additional information on why index `2` is used here

Reply:
does anyone even know? It's not documented anywhere but everyone does it all the same.
Could you suggest a sentence explaining this additional information?

Reply:
Something like

```suggestion
	/* Prepare data for notifications 
	 * attrs[0] is our service declaration
	 * attrs[1] is our characteristic declaration
	 * attrs[2] is our characteristic value
	 *
	 * We store a pointer for the characteristic value as that is the 
	 * value we want to notify later
	 */
	attr = &test_gatt_service.attrs[2];
```

Alternatively we can use `bt_gatt_notify_uuid` to avoid this